e2e/test/scenarios/dashboard/dashboard-card-resizing.cy.spec.js

```diff
@@ -0,0 +1,171 @@
+import {
+  addOrUpdateDashboardCard,
+  editDashboard,
+  getDashboardCards,
+  restore,
+  saveDashboard,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DB_ID } from "e2e/support/cypress_data";
+import { ORDERS, ORDERS_ID, PEOPLE } from "metabase-types/api/mocks/presets";
+import {
+  getDefaultSize,
+  getMinSize,
+} from "metabase/visualizations/shared/utils/sizes";
+
+const MOCK_QUESTION_NAME = "MOCK_QUESTION_NAME";
+
+const commonQuestionFields = {
+  name: MOCK_QUESTION_NAME,
+  query: {
+    "source-table": ORDERS_ID,
+    limit: 10,
+    aggregation: [["count"]],
+  },
+  database: SAMPLE_DB_ID,
+};
+
+// covers table, bar, line, pie, row, area, combo, pivot, funnel, detail, and waterfall questions
+const createMockChartQuestion = vizType => {
+  return {
+    ...commonQuestionFields,
+    query: {
+      ...commonQuestionFields.query,
+      breakout: [["field", ORDERS.CREATED_AT, { "temporal-unit": "minute" }]],
+    },
+    display: vizType,
+  };
+};
+
+// covers scalar, gauge, and progress questions
+const createMockScalarQuestion = vizType => {
+  return {
+    ...commonQuestionFields,
+    display: vizType,
+  };
+};
+
+// covers map questions
+const createMockMapQuestion = () => {
+  return {
+    ...commonQuestionFields,
+    query: {
+      ...commonQuestionFields.query,
+      breakout: [["field", PEOPLE.STATE, { "source-field": ORDERS.USER_ID }]],
+    },
+    display: "map",
+  };
+};
+
+const TEST_DATA = [
+  ...[
+    "table",
+    "bar",
+    "line",
+    "pie",
+    "row",
+    "area",
+    "combo",
+    "pivot",
+    "funnel",
+    "object",
+    "waterfall",
+  ].map(vizType => [vizType, createMockChartQuestion(vizType)]),
+  ...["scalar", "gauge", "progress"].map(vizType => [
+    vizType,
+    createMockScalarQuestion(vizType),
+  ]),
+  ["map", createMockMapQuestion()],
+];
+
+describe("scenarios > dashboard card resizing", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  TEST_DATA.forEach(([vizType, question]) => {
+    it(`should initially display the ${vizType} card with its default size`, () => {
```

Comment:
Looks expensive, could we add a single dashboard with multiple questions with all the display types at once?

Reply:
Yeah I can do that